### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by feuri**
I've added a function to upgrade all VCS based packages. Also added a vim modeline to (almost) every file, because I use vim (hope, I guessed the indent depth (8 chars) right).

### CODE REVIEW

**Comment by ushis** [comment]
Hey thanks, i am using vim too, but prefer 4 chars indent. I will look through this later this evening.

**Comment by feuri** [comment]
That's great. I normally use 4 chars, too.

**Comment by ushis** [comment]
looks goot to me, i'll merge this soon.

**Comment by feuri** [comment]
If we don't use `-V` for `--version`, it sounds good.

**Comment by ushis** [comment]
Im not interrested in a version option at the moment and if we use `-V` for vcs, we still have `-v` for that

File: localrepo/localrepo.py [hunk 1]
```diff
@@ -188,6 +191,34 @@ def aur_upgrade(self):
 
 		return self.add([pkg['uri'] for pkg in updates], True)
 
+	def vcs_upgrade(self):
+		''' Upgrades all VCS packages from the AUR '''
+		Msg.process(_('Updating all VCS packages'))
+
+		try:
+			pkgs = Aur.packages(self.repo.packages)
```
Comment (ushis) [inline localrepo/localrepo.py]: ooh one thing here. we should filter the local packages for vcs names before asking the aur for info. try to keep the request minimal. something like

``` python
vcs = [pkg for pkg in self.repo.packages if re.match('-(?:cvs|svn|hg|darcs|bzr|git)$', pkg)]
#...
Aur.packages(vcs)
```

**Pull request merged**